[FR] Added nautilus extension

DodoLeDev opened the pull request:
#### Hi everyone!
I've added a rudimentary* nautilus extension in order to be able to launch Paper Clip from the explorer.

It is based on a nautilus extension I've first created for the [Collision](https://github.com/GeopJr/Collision/blob/main/nautilus-extension/collision-extension.py) project, then re-used for [Valent](https://github.com/andyholmes/valent/discussions/355#discussioncomment-6410421).

Now it's the turn of Paper Clip! :stuck_out_tongue:

<br><br><br>
_*note: there is still some issues with this script because it was designed for a native installation, and therefore will probably not work with the flatpak version._

sophie-h committed:
doap: Fix listing of online accounts

DodoLeDev committed:
Added nautilus extension

DodoLeDev committed:
Added nautilus extension

Diego-Ivan committed:
doap: Fix listing of online accounts

doap: Fix listing of online accounts

Diego-Ivan commented:
Looks great! In the next few days I will be adding support for opening multiple files in different windows, so it is not necessary to check it is only one file passed.

By the way, I think it is possible to use Flatpak Paper Clip with the extension using `flatpak run --file-forwarding io.github.diegoivan.pdf_metadata_editor FILE_PATH`, but it should be pointed out that it would require changing Paper Clip sandbox permissions.

milotype committed:
Update Croatian translation

milotype committed:
Update hr.po

Diego-Ivan committed:
po: Update croatian translation

Update croatian translation 20230830

Diego-Ivan committed:
window: Use Adw.ToolbarView instead of Gtk.Box

This commit also hides the save, save-as and open-with actions when they
are not activated

Diego-Ivan committed:
details-list: Use Adw.ActionRow with `.property` style class

Removes `DetailRow` as consequence

Diego-Ivan committed:
chore: Minor code style fixes:

* Fix URLs in About Window
* Disconnect all signals in `DocManager`
* Nitpicks with code style

Diego-Ivan committed:
window, document-manager: DocumentManager is no longer a single instance

Diego-Ivan committed:
save-changes-dialog: Implement standalone class for save changes dialog

This class is capable of listing documents and selecting which ones are
going to be saved

Diego-Ivan committed:
window: Port Window to SaveChangesDialog

It also now closes with <Ctrl>W instead of <Ctrl>Q

Diego-Ivan committed:
application: Support multiple windows

The unsaved and selected documents are saved if the user chooses to

Diego-Ivan committed:
window,shortcut-window: Add new keyboard shortcuts

Diego-Ivan committed:
flatpak: Update Poppler dependency, add libtiff dependency

Older libtiff is required by Poppler

Diego-Ivan committed:
application: Support opening multiple files

Ser82-png committed:
Add Russian translation

Diego-Ivan committed:
po: Add Russian translation

Add Russian translation

Diego-Ivan committed:
flatpak: Update Flatpak Manifest to GNOME 45

Diego-Ivan committed:
general: Bump to version 3.5

Diego-Ivan committed:
po: Update PO files and update Spanish translation

Diego-Ivan committed:
chore: Update to GNOME 45 runtime and enhancements

chore: Update to GNOME 45 runtime and enhancements

Diego-Ivan committed:
data: Update screenshots and fix metainfo

Ser82-png committed:
update ru.po

rene-coty committed:
Updated French translation

Diego-Ivan committed:
po: Updated French translation

Updated French translation

Diego-Ivan committed:
po: update ru.po

update ru.po

Alexmelman88 committed:
Update Russian translation

Alexmelman88 committed:
Update ru.po

Diego-Ivan committed:
po: Update Russian translation

Update Russian translation

Commit:
data: update the POTFILES

## Remove untranslatable files
- main.vala
- window.vala

## Add missing file
- keyboard-list.ui

## Order entries alphabetically

Commit:
data: update the POT file

- Add new strings
- Remove unused strings

Commit:
Update Turkish translation

Diego-Ivan committed:
po: Translation related three patches

Translation related three patches

Diego-Ivan committed:
po: Update Spanish Translation

Diego-Ivan committed:
data: Make release description non-translatable

Diego-Ivan committed:
po: Update Spanish Translation, PO fixes

po: Update Spanish Translation, PO fixes

albanobattistella committed:
Update it.po

Alexmelman88 committed:
Update Russian translation

Diego-Ivan committed:
po: Update it.po

Update it.po

Diego-Ivan committed:
po: Update Russian translation

Update Russian translation

Mejans committed:
Added Occitan to list

Mejans committed:
Create Occitan po file

Diego-Ivan committed:
keyword-list: Use <Ctrl>Return to add a keyword

Diego-Ivan committed:
i18n: adds Occitan translation

i18n: adds Occitan translation

Diego-Ivan committed:
shortcut-window: Include add keyword shortcut

Commit:
Update Turkish translation

Small fix for the discard translation

Diego-Ivan committed:
po: Update Turkish translation

Update Turkish translation

Diego-Ivan committed:
keyword-list: Use <Ctrl>Return to add a keyword

keyword-list: Use <Ctrl>Return to add a keyword

Commit:
appdata: Use appstreamcli for appdata validation

appstream-util is obsoleted by appstreamcli.

Commit:
appdata: Update appdata

- Add categories and keywords
- Mark the developer_name as untranslatable
- Add launchable and translation tags

These tags may improve Software Center accessibility.

More information: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-categories

Commit:
po: Update the POT file

Commit:
Update Turkish translation

Diego-Ivan committed:
data: Appdata related patches

Appdata related patches

Diego-Ivan committed:
po: Translation related patches

Translation related patches

Ser82-png committed:
update ru.po

Diego-Ivan committed:
po: update ru.po

update ru.po

Diego-Ivan committed:
general: Release 3.6

Amereyeu committed:
updated czech translation

Amereyeu committed:
updated czech translation

Diego-Ivan committed:
po: updated czech translation

updated czech translation

Diego-Ivan committed:
general: Drop Libportal dependency

Diego-Ivan committed:
general: Drop Libportal dependency

Diego-Ivan committed:
general: Add libexempi as dependency

Also requires libboost

Diego-Ivan committed:
general: Add libexempi as dependency

Also requires libboost

Diego-Ivan committed:
vapi: Add Vala Bindings for libexempi, add as dependency

Diego-Ivan committed:
vapi: Add Vala Bindings for libexempi, add as dependency

Diego-Ivan committed:
application: Start XMP from the Application

Diego-Ivan committed:
application: Start XMP from the Application

Diego-Ivan committed:
thumbnail: Remove I/O operations

We now directly download the data from the Cairo Surface
into a Gdk.MemoryTexture

Diego-Ivan committed:
thumbnail: Remove I/O operations

We now directly download the data from the Cairo Surface
into a Gdk.MemoryTexture

Diego-Ivan committed:
document: Load document using GFile instead of uris

Diego-Ivan committed:
document: Load document using GFile instead of uris

Diego-Ivan committed:
document: Initial support for XMP Metadata

* vapi: Add missing methods to Xmp.File
* document: Support editing XMP metadata

Diego-Ivan committed:
document: Initial support for XMP Metadata

* vapi: Add missing methods to Xmp.File
* document: Support editing XMP metadata

Diego-Ivan committed:
document: Add support to edit Keywords in XMP Metadata

Diego-Ivan committed:
document: Add support to edit Keywords in XMP Metadata

Diego-Ivan committed:
vapi: Add missing classes and methods to libexempi VAPI

* Add Iterator
* Add missing methods in Xmp.Packet

Diego-Ivan committed:
vapi: Add missing classes and methods to libexempi VAPI

* Add Iterator
* Add missing methods in Xmp.Packet

Diego-Ivan committed:
date-row: Notify when the date has been set to null

Diego-Ivan committed:
date-row: Notify when the date has been set to null

Diego-Ivan committed:
document: Support DateTime in XMP Metadata

Diego-Ivan committed:
document: Support DateTime in XMP Metadata

Diego-Ivan committed:
flatpak: Only build the necessary Boost libraries

Considerably reduces build time with Flatpak

Diego-Ivan committed:
flatpak: Only build the necessary Boost libraries

Considerably reduces build time with Flatpak

Diego-Ivan committed:
document: Make saving a document asynchronous

Diego-Ivan committed:
document: Make saving a document asynchronous

Diego-Ivan committed:
exempi: Fix an enum value name in vapi

Diego-Ivan committed:
exempi: Fix an enum value name in vapi

Diego-Ivan committed:
thumbnail: Resize thumbnail before loading it into memory

This considerably improves memory usage, which is reduced from 20-40
MB, depending on the size of the PDF documents. There is still room
for improvement, thought!

Diego-Ivan committed:
thumbnail: Resize thumbnail before loading it into memory

This considerably improves memory usage, which is reduced from 20-40
MB, depending on the size of the PDF documents. There is still room
for improvement, thought!

Diego-Ivan committed:
thumbnailer: Handle high-resolution PDFs differently

Now, to speed things up, high resolution documents will be loaded
in smaller sizes, while smaller thumbnails will be downloaded
directly into memory

Diego-Ivan committed:
thumbnailer: Handle high-resolution PDFs differently

Now, to speed things up, high resolution documents will be loaded
in smaller sizes, while smaller thumbnails will be downloaded
directly into memory

Diego-Ivan committed:
thumbnailer: Load low-resolution images in low memory levels

Diego-Ivan committed:
thumbnailer: Load low-resolution images in low memory levels

Diego-Ivan committed:
thumbnail: Improve thumbnailing (#52)

Diego-Ivan committed:
flatpak: Remove Boost as dependency

Diego-Ivan committed:
flatpak: Remove Boost as dependency

Diego-Ivan committed:
thumbnailer: Remove memory monitor

Diego-Ivan committed:
thumbnailer: Remove memory monitor

Diego-Ivan committed:
Merge branch 'wip/xmp-support' of github.com:Diego-Ivan/Paper-Clip into wip/xmp-support

Diego-Ivan committed:
document: Support MetadataDate XMP property

Diego-Ivan committed:
general: XMP Metadata support (#53)

Diego-Ivan committed:
README: Update listed dependencies

Diego-Ivan committed:
Merge branch 'nautilus-extension' of github.com:DodoLeDev/Paper-Clip into dodoledev-nautilus-extension

Diego-Ivan committed:
nautilus-extension: Use Unicode ellipsis

Diego-Ivan committed:
nautilus-extension: Use Unicode ellipsis

Diego-Ivan committed:
Merge branch 'nautilus-extension' of github.com:DodoLeDev/Paper-Clip into dodoledev-nautilus-extension
